app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -300,4 +336,44 @@ public void showNotificationGroup(long priority) {
                 (NotificationManager) this.getSystemService(Context.NOTIFICATION_SERVICE);
         notificationManager.notify(-5, b.build());
     }
+
+    private Bitmap getIcon(Integer appid) {
+
+        if(appid==-1){
+            return BitmapFactory.decodeResource(getResources(), R.drawable.gotify);
+        }
+
+        Bitmap icon;
+        RequestCreator request;
+        try {
+            request= cache.load(Utils.resolveAbsoluteUrl(settings.url() + "/", appIdMap.get(appid)));
+            icon = request.get();
+        } catch (IOException e) {
```

Comment:
Log the exception with `com.github.gotify.log.Log#e(java.lang.String, java.lang.Throwable)`